_posts/2021-12-15-securing-wildfly-apps-openid-connect.adoc

```diff
@@ -0,0 +1,158 @@
+configuration, there are a couple configuration options that must be specified:
+
+* client-id - This is the client identifier for the application. This is the identifier that
+has been registered with the authorization server.
+
+* provider-url - The URL of the OpenID provider. WildFly will use `provider-url/.well-known/configuration`
+to discover more information about the OpenID provider. As an example, for Keycloak, the `provider-url`
+will look like http://localhost:8080/auth/realms/myrealm.
+
+The rest of the configuration options are optional and are the same as the previous Keycloak
+client adapter.
+
+== OpenID Providers
+
+WildFly 25 and WildFly 26 have been tested with the Keycloak OpenID provider. Although it's now possible
+to configure WildFly to use other OpenID providers, these haven’t been extensively tested yet so the
+use of other OpenID providers should be considered experimental for now and should not be used in a
+production environment yet. Proper support for other OpenID providers will be added in a future WildFly
+release.
+
+== Securing an application using OIDC
+
+In the rest of this post, we'll go through a complete example to see how to secure an application
+deployed to WildFly using OpenID Connect with the Keycloak OpenID provider.
+
+=== Example Project
+
+First, clone the `elytron-examples` repo locally:
+
+[source]
+----
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+cd elytron-examples
+----
+
+We’re going to be looking at the `simple-webapp-oidc` project.
+
+=== Setting up your Keycloak OpenID provider
+
+It's easy to set up Keycloak using Docker. Follow the steps in Keycloak's https://www.keycloak.org/getting-started/getting-started-docker[getting started guide]
+to start Keycloak, create a realm called `myrealm`, create a user called `myuser`,
+and register a client called `myclient`.
+
+After registering our client, `myclient`, we also need to configure valid redirect URIs. Simply click
+on `Clients` and then on `myclient`. In the `Valid Redirect URIs` field, enter http://localhost:8090/simple-webapp-oidc/*.
+
+=== Deploying the app to WildFly
+
+We're going to use the deployment configuration approach for this example, i.e., our application
+contains an https://github.com/wildfly-security-incubator/elytron-examples/blob/master/simple-webapp-oidc/src/main/webapp/WEB-INF/oidc.json[oidc.json] file and its https://github.com/wildfly-security-incubator/elytron-examples/blob/master/simple-webapp-oidc/src/main/webapp/WEB-INF/web.xml#L18-L20[web.xml] file specifies `OIDC` as the `auth-method`.
+
+Now that we've set up our Keycloak OpenID provider, we just need to deploy our application to WildFly.
+
+First, we're going to start our WildFly instance (notice that we're specifying a port offset here
+since our Keycloak instance is already exposed on port 8080):
+
+[source]
+----
+./bin/standalone.sh -Djboss.socket.binding.port-offset=10
```

Comment:
@fjuma Just a detail, but offset should be 10000 here or 10 below

Reply:
The one below is the port itself (so 9990 + 10).

Reply:
@fjuma My mistake. Thanks!